includes/Jobs/RequestSSLCFAddJob.php

```diff
@@ -0,0 +1,412 @@
+					'type' => 'dv',
+					'settings' => [
+						'http2' => 'on',
+						'tls_1_3' => 'on',
+						'min_tls_version' => $tlsVersion,
+					]
+				]
+			] );
+
+			$hostnameId = $response['result']['id'] ?? null;
+
+			if ( !$hostnameId ) {
+				$this->logger->error( 'Failed to create custom hostname for {domain}', [
+					'domain' => $customDomain,
+					'response' => json_encode( $response ),
+				] );
+
+				$this->setLastError( 'Failed to create custom hostname for ' . $customDomain );
+
+				return $response;
+			}
+
+			$this->logger->info( 'Successfully created custom hostname for {domain}, ID: {id}', [
+				'domain' => $customDomain,
+				'id' => $hostnameId,
+			] );
+
+			// Step 2: Poll the hostname status
+			$status = 'pending';
+			$this->logger->debug( 'Polling custom hostname status until active...' );
+
+			while ( $status === 'pending' ) {
+				sleep( 10 );
+
+				$statusResponse = $this->createRequest( '/zones/' . $this->zoneId .
+				 '/custom_hostnames/' . $hostnameId, 'GET' );
+
+				if ( !$statusResponse || !isset( $statusResponse['result'] ) ) {
+					$this->logger->error( 'Failed to retrieve hostname status for {id}', [
+						'id' => $hostnameId,
+					] );
+					$this->setLastError( 'Failed to retrieve status for hostname ID ' . $hostnameId );
+					return $statusResponse;
+				}
+
+				$status = $statusResponse['result']['status'] ?? 'unknown';
+				$errors = $statusResponse['errors'] ?? [];
+				$verificationErrors = $statusResponse['result']['verification_errors'] ?? [];
+
+				$this->logger->debug( 'Hostname status is {status} for {id}', [
+					'status' => $status,
+					'id' => $hostnameId,
+				] );
+
+				if ( $errors ) {
+					$this->logger->error( 'Error encountered while checking hostname status: {errors}', [
+						'errors' => json_encode( $errors ),
+					] );
+					$this->setLastError( 'Error encountered while checking hostname status: ' .
+					json_encode( $errors ) );
+					return $statusResponse;
+				}
+
+				if ( $status === 'pending' && $verificationErrors ) {
+					$this->logger->error( 'Verification failed for hostname {id}', [
+						'id' => $hostnameId,
+						'errors' => json_encode( $verificationErrors ),
+					] );
+					$this->setLastError( 'Verification errors encountered for ' . $customDomain );
+					return $statusResponse;
+				}
+
+				if ( $status != 'active' && $status != 'pending' ) {
+					$this->logger->debug( 'Something went wrong. Status is {status}. Aborting!', [
+						'id' => $hostnameId,
+						'status' => $status,
+					] );
+					$this->setLastError( 'Unexpected status for hostname ID ' . $hostnameId . ': ' . $status );
+					return $statusResponse;
+				}
+			}
+
+			if ( $status === 'active' ) {
+				$this->logger->info( 'Custom hostname {id} is now active.', [
+					'id' => $hostnameId,
+				] );
+			}
+
+			// It's used, I promise!
+			// @phan-suppress-next-line PhanPossiblyUndeclaredVariable
+			return $statusResponse;
+		} catch ( Exception $e ) {
```

Comment:
Is there any more specific error we can catch here? If not it's fine just wondering.